src/main/java/org/matsim/run/rebalancing/WaitingPointsBasedRebalancingStrategy.java

```diff
@@ -0,0 +1,117 @@
+			// when there are idling vehicle that can be rebalanced
+			// we first identify the distribution of the vehicles
+			Map<Id<Link>, MutableInt> waitingPointsOccupancyMap = new HashMap<>();
+			Map<Id<DvrpVehicle>, Id<Link>> vehicleLocationMap = new HashMap<>();
+			waitingPointsCapcityMap.keySet().forEach(linkId -> waitingPointsOccupancyMap.put(linkId, new MutableInt()));
+			for (DvrpVehicle v : fleet.getVehicles().values()) {
+				Schedule s = v.getSchedule();
+				StayTask stayTask = (StayTask) Schedules.getLastTask(s);
+				Link finalStayTaskLink = stayTask.getLink();
+				vehicleLocationMap.put(v.getId(), finalStayTaskLink.getId());
+				if (waitingPointsOccupancyMap.containsKey(finalStayTaskLink.getId())) {
+					waitingPointsOccupancyMap.get(finalStayTaskLink.getId()).increment();
+				}
+			}
+
+			for (DvrpVehicle idleVehicle : idleVehicles) {
+				Id<Link> currentLinkId = vehicleLocationMap.get(idleVehicle.getId());
+				if (!waitingPointsOccupancyMap.containsKey(currentLinkId)) {
+					// vehicle is not at any waiting point -> go to the nearest waiting point that is not yet full
+					Link nearestAvailableWaitingPointLink = findNearestAvailableWaitingPoint(currentLinkId, waitingPointsOccupancyMap);
+					relocations.add(new Relocation(idleVehicle, nearestAvailableWaitingPointLink));
+					waitingPointsOccupancyMap.get(nearestAvailableWaitingPointLink.getId()).increment();
+				} else if (waitingPointsOccupancyMap.get(currentLinkId).intValue() > waitingPointsCapcityMap.get(currentLinkId)) {
+					// it is stopping at a waiting point, but there is no extra space -> go to next waiting point that is not yet full
+					Link nearestAvailableWaitingPointLink = findNearestAvailableWaitingPoint(currentLinkId, waitingPointsOccupancyMap);
+					relocations.add(new Relocation(idleVehicle, nearestAvailableWaitingPointLink));
+					waitingPointsOccupancyMap.get(nearestAvailableWaitingPointLink.getId()).increment();
+					waitingPointsOccupancyMap.get(currentLinkId).decrement();
+				}
+				// else, stay where it is now.
+			}
+		}
+		return relocations;
+	}
+
+	private Link findNearestAvailableWaitingPoint(Id<Link> currentLinkId, Map<Id<Link>, MutableInt> waitingPointsOccupancyMap) {
+		double shortestDistance = Double.POSITIVE_INFINITY;
+		Link nearestWaitingPoint = null;
+
+		for (Id<Link> waitingPoint : waitingPointsOccupancyMap.keySet()) {
+			if (waitingPointsOccupancyMap.get(waitingPoint).intValue() < waitingPointsCapcityMap.get(waitingPoint)) {
+				double distance = CoordUtils.calcEuclideanDistance(network.getLinks().get(currentLinkId).getToNode().getCoord(),
+					network.getLinks().get(waitingPoint).getToNode().getCoord());
+				if (distance < shortestDistance) {
+					nearestWaitingPoint = network.getLinks().get(waitingPoint);
+					shortestDistance = distance;
+				}
+			}
+		}
+		Preconditions.checkArgument(nearestWaitingPoint != null, "Not able to find nearest waiting point! Please make sure the sum capacity" +
```

Comment:
maybe we should clarify to the user in form of a JvaDoc for the RebalancingStrategy, that enough capacity needs to be provided. At least when we move the code into matsim-libs

Reply:
Aternatively to aborting, one could argue to leave the vehicle where it is, or to speculatively send it to the nearest waiting point, hoping that that is cleared in the meantime.

Reply:
Okay, I will do the refactoring. 

I will add the using start location as the depot. That is indeed more handy for us.